### PR TITLE
add mit side import and entity mapping file

### PR DESCRIPTION
**Pull request by Tranway1**

### CODE REVIEW

File: askem_extractions/importers/mit.py
Comment (enoriega) [inline askem_extractions/importers/mit.py:1]: @Tranway1 , in general this function looks good. Before merging we need to decouple the parts that read the MIT extractions from the code that merges them with AZ extractions.

The reason for this is that it is plausible that you want to load MIT extractions without having AZ extractions and the mapping and if this function requires them, you can't use it.

The pseudo code of this workflow could be something like this:
```python
mit_extractions = import_mit(Path("mit_data.json"))
az_extractions = import_arizona(Path("az_data.json"))

merged_extractions = merge_collections(mit_extractions, az_extractions, Path("mapping.txt"))
```

I left comments with my suggestions on the relevant lines

File: askem_extractions/importers/mit.py [hunk 1]
```diff
@@ -1,8 +1,64 @@
...
 from pathlib import Path
 
-from askem_extractions.data_model import ExtractionsCollection
+from askem_extractions.data_model import ExtractionsCollection, DKGConcept, Dataset, DataColumn
 
 
-def import_mit(path: Path) -> ExtractionsCollection:
-    # TODO Implement MIT's output format importer
-    ...
+def import_mit(a_path: Path, m_path: Path, map_path: Path) -> ExtractionsCollection:
```
Comment (enoriega) [inline askem_extractions/importers/mit.py]: I suggest breaking up the functionality into two functions:

- `import_mit` will only be responsible to parse your output data format (`m_path`)
- A new function: `merge_collections` that takes AZ and MIT's imported data and merges it using the map (`a_path` and `map_path`)

File: askem_extractions/importers/mit.py [hunk 2]
```diff
@@ -1,8 +1,64 @@
...
+    # Extract the data from json file
+    from_json = ExtractionsCollection.from_json(a_path)
```
Comment (enoriega) [inline askem_extractions/importers/mit.py:10]: This could go in `merge_collections`

File: askem_extractions/importers/mit.py [hunk 3]
```diff
@@ -1,8 +1,64 @@
...
+
+    with open(m_path, "r") as file_a:
+        data_m = json.load(file_a)
+
+    # Load mapping file
+    with open(map_path, "r") as mapping_file:
+        mappings = mapping_file.readlines()
```
Comment (enoriega) [inline askem_extractions/importers/mit.py:17]: These could go in `merge_collections`

File: askem_extractions/importers/mit.py [hunk 4]
```diff
@@ -1,8 +1,64 @@
...
+                                id=term[2],
+                            )
+                            column = DataColumn(
+                                name=term[1],
+                                id=str(term[2])+"-"+str(term[0]),
+                                dataset=dataset,
+                            )
+                            vs.variable.column.append(column)
+                    # if entry_a["equation_annotations"] is empty
+    from_json.save_json("TA1-integration.json")
```
Comment (enoriega) [inline askem_extractions/importers/mit.py:61]: This could be the return value of `merge_collections`

File: askem_extractions/importers/mit.py [hunk 5]
```diff
@@ -1,8 +1,64 @@
...
+                            dataset = Dataset(
+                                name=term[3],
+                                id=term[2],
+                            )
+                            column = DataColumn(
+                                name=term[1],
+                                id=str(term[2])+"-"+str(term[0]),
+                                dataset=dataset,
+                            )
+                            vs.variable.column.append(column)
```
Comment (enoriega) [inline askem_extractions/importers/mit.py:59]: This block could probably be the only part remaining in this function and the outer loop could go on the `merge_collections`

File: askem_extractions/importers/mit.py [hunk 6]
```diff
@@ -1,8 +1,64 @@
...
+        key, value = mapping.strip().split(": ")
+        mapping_dict[key] = value.strip('"').strip(",")
+
+    for vs in from_json.variable_statements:
+        entry_b_id = vs.id
+        print(entry_b_id)
+        if entry_b_id in mapping_dict.values():
+            print("Found mapping")
+            # Get the corresponding key (id from data_a) and find the entry in data_a
+            entry_a_id = [k for k, v in mapping_dict.items() if v == entry_b_id][0]
```
Comment (enoriega) [inline askem_extractions/importers/mit.py:31]: This outer block should probably go in `merge_collections`

File: askem_extractions/importers/mit.py [hunk 7]
```diff
@@ -1,8 +1,64 @@
...
+                                id=str(term[2])+"-"+str(term[0]),
+                                dataset=dataset,
+                            )
+                            vs.variable.column.append(column)
+                    # if entry_a["equation_annotations"] is empty
+    from_json.save_json("TA1-integration.json")
+
+
+if __name__ == "__main__":
+    import_mit(Path("../../examples/temp.json"), Path("../../examples/mit_extraction.json"), Path("../../examples/mapping.txt"))
```
Comment (enoriega) [inline askem_extractions/importers/mit.py]: Can you delete this block? It is already included in `usage.py`

**Review by enoriega — CHANGES_REQUESTED** [review]

**Comment by Tranway1** [comment]
Please check the refactored code

**Review by enoriega — APPROVED** [review]
Looks good to me!

**Comment by enoriega** [comment]
@Tranway1 Thanks! This is merged into main. I think we are ready to share it with TA4 folks

**Pull request merged**